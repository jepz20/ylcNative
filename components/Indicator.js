import React from 'react'
import { View, StyleSheet } from 'react-native'

const Indicator = ({ style }) => {
  return (
    <View style={[styles.indicator, style]} />
  )
}

const styles = StyleSheet.create({
  indicator: {
    borderRadius: 50,
    backgroundColor: 'grey',
    height: 30,
    width: 30,
    borderWidth: 1,
    elevation: 2,
    borderColor: '#000'
  }
})

export { Indicator }
