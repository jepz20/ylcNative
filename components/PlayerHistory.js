// @flow

import * as React from 'react'
import {
  View,
  Text,
  StyleSheet,
  ScrollView,
  TouchableWithoutFeedback
} from 'react-native'
import type { LogMap, LogList } from '../types/match'
import { mapDueslPerMatch } from '../core'
import { NO_LOG_CHARACTER } from '../constants'
import { Map } from 'immutable'
import type { ToggleCalculatorVisibility } from '../actions/calculator'

const POSITIVE = 'green'
const NEGATIVE = 'red'
type Props = {
  logs: LogMap,
  playerId: string,
  currentDuel: string,
  toggleCalculatorVisibility: ToggleCalculatorVisibility
}

const PlayerHistory = ({
  logs,
  playerId,
  currentDuel = '1',
  toggleCalculatorVisibility
}: Props) => {
  const currentLog: ?LogList = logs.get(String(currentDuel))
  const renderLog = (currentLog: LogList) =>
    currentLog
      .reverse()
      .map(({ operationValue, playerId: logPlayerId, previousPoints, id }) => {
        if (logPlayerId !== playerId) return null
        return (
          <View style={[styles.logContainer]} key={id}>
            <Text
              style={[
                styles.operationValue,
                { color: operationValue > 0 ? POSITIVE : NEGATIVE }
              ]}
            >
              {operationValue > 0 && '+'}
              {operationValue}
            </Text>
            <Text style={[styles.currentPoints]}>{previousPoints}</Text>
          </View>
        )
      })

  const renderNoLogs = (noLogChar: string) =>
    mapDueslPerMatch(index => (
      <View style={styles.logContainer} key={index}>
        <Text style={styles.noLog}>{noLogChar}</Text>
      </View>
    ))

  return (
    <ScrollView contentContainerStyle={styles.container}>
      <TouchableWithoutFeedback
        onPress={() => toggleCalculatorVisibility(playerId)}
      >
        <View>
          {currentLog ? renderLog(currentLog) : renderNoLogs(NO_LOG_CHARACTER)}
        </View>
      </TouchableWithoutFeedback>
    </ScrollView>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start'
  },
  logContainer: {
    marginTop: 10,
    marginBottom: 10,
    flexDirection: 'row',
    justifyContent: 'center'
  },
  operationValue: {
    alignSelf: 'flex-end',
    fontSize: 15,
    marginRight: 10,
    marginBottom: 5,
    flex: 1,
    textAlign: 'right'
  },
  currentPoints: {
    fontSize: 30,
    flex: 2,
    textDecorationLine: 'line-through',
    textDecorationStyle: 'solid',
    color: 'grey'
  },
  noLog: {
    fontSize: 30,
    color: 'grey',
    alignSelf: 'flex-end'
  }
})

PlayerHistory.defaultProps = {
  logs: Map()
}

export { PlayerHistory }
