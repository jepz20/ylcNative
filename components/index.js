// @flow

export * from './Indicator'
export * from './PointsCounter'
export * from './PlayerHistory'
export * from './PlayerDetail'
export * from './Row'
export * from './Button'
export * from './TextButton'
export * from './IconButton'
