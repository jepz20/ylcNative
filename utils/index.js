// @flow

import { Range } from 'immutable'

export type FillFunction = (index: string) => any

export const mapRangeOfLength = (length: number) => (
  fillFunction: FillFunction
) => {
  if (length < 1) {
    throw new Error('length should be greater than 1')
  }
  return Range(1, length + 1).map((index: number) =>
    fillFunction(index.toString())
  )
}

export const parsePoints = (value: ?string): number =>
  value ? parseInt(value, 10) : 0
