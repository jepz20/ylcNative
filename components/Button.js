// @flow

import type { StyleObj } from 'react-native/Libraries/StyleSheet/StyleSheetTypes'
import { renderEl } from '../utils'
import * as React from 'react'
import { TouchableHighlight, StyleSheet } from 'react-native'
type Props = {
  children: React.Node,
  style?: StyleObj
}
const Button: React.StatelessFunctionalComponent<Props> = ({
  style,
  children,
  ...props
}) => {
  if (!children) return null

  return (
    <TouchableHighlight
      style={[styles.container, style]}
      underlayColor='rgba(0, 0, 0, 0.4)'
      {...props}
    >
      {renderEl(children)}
    </TouchableHighlight>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  text: {
    color: '#fff',
    fontSize: 40
  }
})

export { Button }
