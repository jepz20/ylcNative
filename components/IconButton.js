// @flow

import * as React from 'react'
import { Button } from './Button'
import { Entypo } from '@expo/vector-icons'

type Props = {
  name: string,
  size: number,
  color: string
}

const IconButton = ({ name, size, color, ...props }: Props) => {
  return (
    <Button {...props}>
      <Entypo name={name} size={size} color={color} />
    </Button>
  )
}

IconButton.defaultProps = {
  size: 40,
  color: '#FFF'
}

export { IconButton }
