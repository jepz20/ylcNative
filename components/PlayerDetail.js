import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { Indicator, PointsCounter } from '../components'
import isEmpty from 'lodash/isEmpty'
import { createArray } from '../utils'

const LIFE_POINTS_BACKGROUND = 'green'

const PlayerDetail = (props) => {
  const getIndicatorStyle = (winner, player) => {
    if (!winner) {
      return null
    }
    if (winner === 'tie') {
      return styles.indicatorTie
    } else if (winner === player) {
      return styles.indicatorWin
    } else {
      return styles.indicatorLoose
    }
  }

  const renderResultsIndicator = () => {
    const { matchResults, settings, id } = props
    if (isEmpty(matchResults)) {
      return null
    }

    return createArray(settings.duelsPerMatch).map((_, index) => {
      const { winner } = matchResults[index + 1] || {}
      const style = getIndicatorStyle(winner, id)
      return <Indicator key={index} style={style} />
    })
  }

  const { currentPoints, name } = props
  return (
    <View style={[styles.container, props.outerContainerStyle]}>
      <View style={styles.resultsContainer}>
        {renderResultsIndicator()}
      </View>
      <PointsCounter value={currentPoints} />
      <View style={styles.playerNameContainer}>
        <Text style={styles.playerNameText}>
          {name}
        </Text>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: LIFE_POINTS_BACKGROUND,
    borderColor: '#000',
    borderWidth: 1
  },
  resultsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    height: 35,
    marginTop: 10
  },
  indicatorWin: {
    backgroundColor: 'green'
  },
  indicatorLoose: {
    backgroundColor: 'red'
  },
  indicatorTie: {
    backgroundColor: 'yellow'
  },
  playerNameContainer: {
    height: 40,
    justifyContent: 'flex-end',
    marginLeft: 5,
    marginBottom: 5
  },
  playerNameText: {
    fontSize: 17,
    fontWeight: 'bold',
    color: '#fff'
  }
})

export { PlayerDetail }
