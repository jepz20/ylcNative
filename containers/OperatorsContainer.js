// @flow

import * as React from 'react'
import { View, StyleSheet } from 'react-native'
import { TextButton } from '../components'
import { connect } from 'react-redux'
import * as actions from '../actions'
import type { ToggleCalculatorVisibility, Reset } from '../actions/calculator'
import type { StateRE } from '../reducers/calculator'
import type {
  AddPoints,
  SubstractPoints,
  HalfPoints,
  Draw,
  ScoopMatch
} from '../actions/match'
import { getId } from '../utils'

type Props = {
  player: string,
  calculator: StateRE,
  addPoints: AddPoints,
  substractPoints: SubstractPoints,
  halfPoints: HalfPoints,
  scoopMatch: ScoopMatch,
  draw: Draw,
  toggleCalculatorVisibility: ToggleCalculatorVisibility,
  reset: Reset
}
const OperatorsContainerWithoutRedux = ({
  player,
  calculator,
  addPoints,
  substractPoints,
  halfPoints,
  scoopMatch,
  draw,
  toggleCalculatorVisibility,
  reset
}: Props) => {
  const applyOperation = (type, player) => {
    switch (type) {
      case 'scoop':
        scoopMatch(player, getId())
        break
      case 'draw':
        draw(getId())
        break
      case 'halfPoints':
        halfPoints(player, getId())
        break
      case 'add':
        addPoints(calculator.value, player, getId())
        break
      case 'substract':
        substractPoints(calculator.value, player, getId())
        break
    }
    toggleCalculatorVisibility(null)
    reset()
  }

  return (
    <View style={styles.container}>
      <View style={[{ flex: 1 }]}>
        <TextButton
          onPress={() => applyOperation('scoop', player)}
          value='SCOOP'
          name='scoop'
          size='small'
        />
      </View>
      <View style={[styles.row, { flex: 1 }]}>
        <TextButton
          onPress={() => applyOperation('draw', player)}
          value='DRAW'
          size='small'
          name='draw'
        />
        <TextButton
          onPress={() => applyOperation('halfPoints', player)}
          value='1/2'
          size='small'
          name='halfPoints'
        />
      </View>
      <View style={[styles.row, { flex: 2 }]}>
        <TextButton
          onPress={() => applyOperation('add', player)}
          value='+'
          size='large'
          name='add'
        />
        <TextButton
          onPress={() => applyOperation('substract', player)}
          value='-'
          size='large'
          name='substract'
        />
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'green'
  },
  row: {
    flexDirection: 'row'
  }
})

const mapStateToProps = ({ calculator }) => ({
  calculator
})

const OperatorsContainer = connect(mapStateToProps, actions)(
  OperatorsContainerWithoutRedux
)

export { OperatorsContainer }
