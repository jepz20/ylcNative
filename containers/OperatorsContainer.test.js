import React from 'react'
import 'react-native'
import PropTypes from 'prop-types'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import { Record } from 'immutable'
import { OperatorsContainer } from './OperatorsContainer'
import { TextButton } from '../components/index'
import * as Utils from '../utils'

const CalculatorRE = Record({
  currentPlayer: 1,
  visible: true,
  value: '100'
})

const mockStore = configureStore([])({ calculator: CalculatorRE() })
function testProvider () {
  class TestProvider extends React.Component {
    getChildContext () {
      return { store: mockStore }
    }

    render () {
      return this.props.children
    }
  }

  TestProvider.childContextTypes = {
    store: PropTypes.object
  }
  return TestProvider
}

describe('OperatorsContainer', () => {
  const TestProvider = testProvider()
  Utils.getId = jest.fn(() => 1)

  test('renders with no props', () => {
    const operatorsContainer = renderer
      .create(
        <TestProvider>
          <OperatorsContainer />
        </TestProvider>
      )
      .toJSON()
    expect(operatorsContainer).toMatchSnapshot()
  })

  test('renders correctly', () => {
    const TestProvider = testProvider()
    const operatorsContainer = renderer
      .create(
        <TestProvider>
          <OperatorsContainer player='1' />
        </TestProvider>
      )
      .toJSON()
    expect(operatorsContainer).toMatchSnapshot()
  })

  test('scoops a duel', () => {
    const operatorsContainer = renderer.create(
      <TestProvider>
        <OperatorsContainer player='1' />
      </TestProvider>
    ).root
    const scoopButton = operatorsContainer
      .findAllByType(TextButton)
      .filter(it => it.props.name === 'scoop')[0]
    scoopButton.props.onPress()
    expect(mockStore.getActions()).toMatchSnapshot()
  })

  test('draws a duel', () => {
    const operatorsContainer = renderer.create(
      <TestProvider>
        <OperatorsContainer player='1' />
      </TestProvider>
    ).root
    const drawButton = operatorsContainer
      .findAllByType(TextButton)
      .filter(it => it.props.name === 'draw')[0]
    drawButton.props.onPress()
    expect(mockStore.getActions()).toMatchSnapshot()
  })

  test('add of a player', () => {
    const operatorsContainer = renderer.create(
      <TestProvider>
        <OperatorsContainer player='1' />
      </TestProvider>
    ).root
    const halfPointsButton = operatorsContainer
      .findAllByType(TextButton)
      .filter(it => it.props.name === 'halfPoints')[0]
    halfPointsButton.props.onPress()
    expect(mockStore.getActions()).toMatchSnapshot()
  })

  test('adds points to a player', () => {
    const operatorsContainer = renderer.create(
      <TestProvider>
        <OperatorsContainer player='1' />
      </TestProvider>
    ).root
    const addButton = operatorsContainer
      .findAllByType(TextButton)
      .filter(it => it.props.name === 'add')[0]
    addButton.props.onPress()
    expect(mockStore.getActions()).toMatchSnapshot()
  })

  test('substract points to a player', () => {
    const operatorsContainer = renderer.create(
      <TestProvider>
        <OperatorsContainer player='1' />
      </TestProvider>
    ).root
    const addButton = operatorsContainer
      .findAllByType(TextButton)
      .filter(it => it.props.name === 'substract')[0]
    addButton.props.onPress()
    expect(mockStore.getActions()).toMatchSnapshot()
  })
})
