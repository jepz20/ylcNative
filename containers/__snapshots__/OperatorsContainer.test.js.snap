// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperatorsContainer add of a player 1`] = `
Array [
  Object {
    "payload": Object {
      "logId": 1,
      "player": "1",
    },
    "type": "@match/scoop_match",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
  Object {
    "payload": Object {
      "logId": 1,
    },
    "type": "@draw",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
  Object {
    "payload": Object {
      "logId": 1,
      "player": "1",
    },
    "type": "@match/half_points",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
]
`;

exports[`OperatorsContainer adds points to a player 1`] = `
Array [
  Object {
    "payload": Object {
      "logId": 1,
      "player": "1",
    },
    "type": "@match/scoop_match",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
  Object {
    "payload": Object {
      "logId": 1,
    },
    "type": "@draw",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
  Object {
    "payload": Object {
      "logId": 1,
      "player": "1",
    },
    "type": "@match/half_points",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
  Object {
    "payload": Object {
      "logId": 1,
      "player": "1",
      "points": "100",
    },
    "type": "@match/add_points",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
]
`;

exports[`OperatorsContainer draws a duel 1`] = `
Array [
  Object {
    "payload": Object {
      "logId": 1,
      "player": "1",
    },
    "type": "@match/scoop_match",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
  Object {
    "payload": Object {
      "logId": 1,
    },
    "type": "@draw",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
]
`;

exports[`OperatorsContainer renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "green",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        SCOOP
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        DRAW
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        1/2
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "flex": 2,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
            },
            Object {
              "fontSize": 100,
            },
          ]
        }
      >
        +
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
            },
            Object {
              "fontSize": 100,
            },
          ]
        }
      >
        -
      </Text>
    </View>
  </View>
</View>
`;

exports[`OperatorsContainer renders with no props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "green",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        SCOOP
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        DRAW
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        1/2
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "flex": 2,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
            },
            Object {
              "fontSize": 100,
            },
          ]
        }
      >
        +
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 40,
            },
            Object {
              "fontSize": 100,
            },
          ]
        }
      >
        -
      </Text>
    </View>
  </View>
</View>
`;

exports[`OperatorsContainer scoops a duel 1`] = `
Array [
  Object {
    "payload": Object {
      "logId": 1,
      "player": "1",
    },
    "type": "@match/scoop_match",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
]
`;

exports[`OperatorsContainer substract points to a player 1`] = `
Array [
  Object {
    "payload": Object {
      "logId": 1,
      "player": "1",
    },
    "type": "@match/scoop_match",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
  Object {
    "payload": Object {
      "logId": 1,
    },
    "type": "@draw",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
  Object {
    "payload": Object {
      "logId": 1,
      "player": "1",
    },
    "type": "@match/half_points",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
  Object {
    "payload": Object {
      "logId": 1,
      "player": "1",
      "points": "100",
    },
    "type": "@match/add_points",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
  Object {
    "payload": Object {
      "logId": 1,
      "player": "1",
      "points": "100",
    },
    "type": "@substract_points",
  },
  Object {
    "payload": Object {
      "player": null,
    },
    "type": "@toggle_calculator_visibility",
  },
  Object {
    "type": "@rest_calculator_value",
  },
]
`;
